core/src/Realm/UserRealmDiscoveryResponse.cs

```diff
@@ -0,0 +1,67 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+using Microsoft.Identity.Core.Helpers;
+using Microsoft.Identity.Core.Http;
+
+namespace Microsoft.Identity.Core.Realm
+{
+    [DataContract]
+    internal sealed class UserRealmDiscoveryResponse
+    {
+        [DataMember(Name = "ver")]
+        public string Version { get; set; }
+
+        [DataMember(Name = "account_type")]
+        public string AccountType { get; set; }
+
+        [DataMember(Name = "federation_protocol")]
+        public string FederationProtocol { get; set; }
+
+        [DataMember(Name = "federation_metadata_url")]
+        public string FederationMetadataUrl { get; set; }
+
+        [DataMember(Name = "federation_active_auth_url")]
+        public string FederationActiveAuthUrl { get; set; }
+
+        [DataMember(Name = "cloud_audience_urn")]
+        public string CloudAudienceUrn { get; set; }
+
+        internal static async Task<UserRealmDiscoveryResponse> CreateByDiscoveryAsync(string userRealmUri, string userName, RequestContext requestContext)
+        {
+            var msg = "Sending request to userrealm endpoint.";
+            requestContext.Logger.Info(msg);
+            requestContext.Logger.InfoPii(msg);
+            var httpResponse = await HttpRequest.SendGet(
+                new UriBuilder(userRealmUri + userName + "?api-version=1.0").Uri, null, requestContext).ConfigureAwait(false);
```

Comment:
>userRealmUri + userName + "?api-version=1.0" [](start = 31, length = 44)

consider using string interpolation (or string.format with verbatim, @)
reference doc: [https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/strings/](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/strings/)

Reply:
marking as won't fix.

---
In reply to: [201545154](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1118#discussion_r201545154) [](ancestors = 201545154)